src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy

```diff
@@ -29,9 +30,12 @@ import org.gradle.api.tasks.Delete
 import org.gradle.api.tasks.Exec
 import org.gradle.api.tasks.bundling.Zip
 
+import java.util.regex.Pattern
+
 class PalantirDockerPlugin implements Plugin<Project> {
 
     private static final Logger log = Logging.getLogger(PalantirDockerPlugin.class)
+    private static final Pattern labelKeyPattern = Pattern.compile("^[a-z0-9.-]*\$")
```

Comment:
Without escaping it, it fails to compile with `illegal string body character after dollar sign`, which I believe is due to Groovy string replacement. Replacing the double quotes with single quotes fixes this.